nemo_skills/inference/eval/bfcl.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import sys
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import field
+
+import hydra
+import re   
+from dataclasses import asdict, field
+
+from nemo_skills.inference.eval.bfcl_utils import convert_to_function_call, execute_multi_turn_func_call, is_empty_execute_response, MAXIMUM_STEP_LIMIT, DEFAULT_USER_PROMPT_FOR_ADDITIONAL_FUNCTION_FC
+from nemo_skills.inference.generate import GenerateSolutionsConfig, GenerationTask, InferenceConfig
+from nemo_skills.inference.model import server_params
+from nemo_skills.utils import get_help_message, get_logger_name, nested_dataclass, setup_logging
+
+from nemo_skills.dataset.bfcl.utils import func_doc_language_specific_pre_processing, convert_to_tool
+
+LOG = logging.getLogger(get_logger_name(__file__))
+
+
+@nested_dataclass(kw_only=True)
+class BFCLGenerationConfig(GenerateSolutionsConfig):
+    """BFCL benchmark generation."""
+
+    # Inheritance was converting these dataclasses to dicts, so to be on the safe side we override them
+    inference: InferenceConfig = field(default_factory=InferenceConfig)  # LLM call parameters
+    # Inference server configuration {server_params}
+    server: dict = field(default_factory=dict)
+
+    prompt_config: str = "generic/default"
+    prompt_template: str = "default-base"
+
+    thinking_begin: str = "<think>"
```

Comment:
as these are defaults that match the parent class, you don't have to list them explicitly. So can keep remove_thinking but remove the other two